app/assets/javascripts/discourse/dialects/code_dialect.js

```diff
@@ -33,6 +33,7 @@ function codeFlattenBlocks(blocks) {
 Discourse.Dialect.replaceBlock({
   start: /^`{3}([^\n\[\]]+)?\n?([\s\S]*)?/gm,
   stop: /^```$/gm,
+  without_leading: /\[quote/gm, //if leading text contains a quote this should not match
```

Comment:
in JS we prefer camelCase for attributes, unless of course we're referring to something back from the server that came in the underscored format :)